src/keep_gpu/single_gpu_controller/rocm_gpu_controller.py

```
@@ -67,14 +71,48 @@ def keep(self) -> None:
+            self._thread.join(timeout=1.0)
+            self._thread = None
+            self._stop_evt = None
+            self._shutdown_rocm_smi()
+            raise startup_errors[0]
         logger.info("rank %s: ROCm keep thread started", self.rank)
 
+    def _shutdown_rocm_smi(self) -> None:
+        if self._rocm_smi:
+            try:
+                self._rocm_smi.rsmi_shut_down()
+            except Exception as exc:  # pragma: no cover - best effort
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)